lib/core/utils/select.js

```diff
@@ -73,24 +67,68 @@ function pushNode(result, nodes, context) {
 	return result;
 }
 
+/**
+ * returns true if any of the nodes in the list is a parent of another node in the list
+ * @param {Array} the array of include nodes
+ * @return {Boolean}
+ */
+function hasOverlappingIncludes(includes) {
+	let list = includes.slice();
+	while (list.length > 1) {
+		let last = list.pop();
+		if (list[list.length - 1].actualNode.contains(last.actualNode)) {
+			return true;
+		}
+	}
+	return false;
+}
+
 /**
  * Selects elements which match `selector` that are included and excluded via the `Context` object
  * @param  {String} selector  CSS selector of the HTMLElements to select
  * @param  {Context} context  The "resolved" context object, @see Context
  * @return {Array}            Matching virtual DOM nodes sorted by DOM order
  */
 axe.utils.select = function select(selector, context) {
+	//jshint maxstatements:20
 	'use strict';
 
 	var result = [], candidate;
-	for (var i = 0, l = context.include.length; i < l; i++) {
+	if (!Array.isArray(context.include)) {
+		context.include = Array.from(context.include);
+	}
+	context.include.sort(axe.utils.nodeSorter); // ensure that the order of the include nodes is document order
```

Comment:
We should enforce array type and sort the include as part of `normalizeContext` instead. That way this function doesn't have any side effects that can cause bugs later, and we don't have to sort more than once.
https://github.com/dequelabs/axe-core/blob/develop/lib/core/base/context.js#L72

Reply:
We would have to do it in the Context constructor because include is modified after `normalizeContext`. I can make that change.